src/api/search/src/routes/query.js

```diff
@@ -13,4 +13,16 @@ router.get('/', validateQuery, async (req, res, next) => {
   }
 });
 
+// route for advanced
+router.get('/advanced', validateQuery, async (req, res, next) => {
```

Comment:
Per your discussion in the description of this PR, I don't think that there's any point maintaining the existing search endpoint separate from the new advanced one, since the latter is a always special case of the former.  Let's have only one search endpoint, drop the `advanced` bit, and either pass the `postText` or `author` fields as an indication of which we are doing.  Later the front-end can be more precise.  Please file a follow-up issue on this, detailing what you think needs to happen for that to work, so we don't lose track of it here.

Reply:
We can't remove the existing one in this PR, but the one I was working on with the front end does work on removing it - we have to go through and refactor the react context for this change.

Reply:
Then we need an issue filed to harmonize these into a single route.  Can you file that?

Reply:
filed: https://github.com/Seneca-CDOT/telescope/issues/2618